### DISCUSSION TITLE
YouTube new Ad Policy

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#3322] [General]
📺 YouTube finds yet another way to push ads through

It's been a solid year since YouTube declared a de facto war on ad blockers. They've tried many different things, from mere pop-ups that inform viewers that "ad blockers are not allowed", to warnings with countdown clocks, to halting playback after three videos watched with an ad blocker on. But ad blockers have been largely able to keep up and withstand all the punches YouTube has been throwing at them.

There's now a new escalation. The media giant is apparently testing server-side ad injection. Unlike traditional methods, this technique of injecting ads into the code right on the server makes it impossible to distinguish them from content with regular ad-blocking tools.

While this technique poses quite a challenge for ad blockers, there is no reason to wave the white flag. What does anyone think about this?

**Comment by Yuki2718** [comment]
Nothing new, we have been aware of this far before any media making the fuss which was apparently by a tweet of our colleague devolopping SponsorBlock after we talked about that. At the time it had already been common knowledge in the community. We already addressed this and now yt countered our fix and we're going to re-counter.

**Comment by uBlock-user** [comment]
YT already inserts ads through SSAI, nothing new.

**Comment by ghost** [comment]
Can this be fixed?
Some users are having problems

**Reply by Yuki2718** [reply]
Users on Chromium-based browser will see these ads because we disabled the filter for server-side ads on Chromium to fix breakage by Trusted Types policy yt is applying for selected user. A fix for this will be available in the next uBO build. Other than that, yt added counter to the currrent fix for some selected user, so if you're unlucky you might see server-side ads even on Firefox. We will fix this soon.

**Reply by ghost** [reply]
> Users on Chromium-based browser will see these ads because we disabled the filter for server-side ads on Chromium to fix breakage by Trusted Types policy yt is applying for selected user. A fix for this will be available in the next uBO build. Other than that, yt added counter to the currrent fix for some selected user, so if you're unlucky you might see server-side ads even on Firefox. We will fix this soon.

Will we no longer be able to block ads in Chromium?

**Reply by Yuki2718** [reply]
> A fix for this will be available in the next uBO build.
